baseapp/abci.go

```diff
@@ -304,6 +304,17 @@ func (app *BaseApp) DeliverTx(req abci.RequestDeliverTx) abci.ResponseDeliverTx
 	}()
 
 	gInfo, result, anteEvents, err := app.runTx(runTxModeDeliver, req.Tx)
+	// cap gas used at gas wanted + 1, to reduce application sensitivity to gas when were exceeding the limit
+	// cref: https://github.com/cosmos/cosmos-sdk/issues/12788
+	// TODO: This should become a sentinel value, with ABCI updates.
+	// TODO: In the future, the entire gas wanted & gas used being communicated to Tendermint is a bad design
```

Comment:
I don't understand these TODOs, especially the first one. The 2nd one I sort of understand, but that's probably not gonna happen any time soon.

I feel like these will honestly never be addressed.